### DISCUSSION TITLE
Solving crowded quests

### DISCUSSION THREAD

**Discussion by timothywashere** [streetcomplete/StreetComplete#3340] [Ideas]
Hi all,
Still loving Street complete.

A lot of times when I am doing quests on small terraced houses, walking down the street, I want to number them all in turn because using the + and - buttons is really convenient and if I do them out of order I lose track and have to stop. 

The app hides some of the quests I want to solve because they are crowded by the quests I don't want to solve yet, as I think you can see from the screenshot below. 

![Screenshot_2021_10_02_17_24_27](https://user-images.githubusercontent.com/87654968/135933805-c451a02e-dcef-494f-8ee3-f7d9d9f6e0b2.jpg)

I don't think there is any way around this. Sometimes I give a quest a fake answer just so I can then solve the one I want and then use the undo function to get rid of the fake one. This often happens at night when I can only see a house number on one house and so can't fill in the neighbouring houses.

I thought it would be nice if I could press the small circle that is shown for crowded quests and it would let me solve the quest there anyway. 

Anyone else have this problem or got any other solutions?
I think maybe in other countries houses are a lot bigger, but the UK is full of these Victorian terraces

**Comment by mnalis** [comment]
- Each quest has `Other answers` / `can't say` / `No, just hide`, which should remove the quest that you don't want to solve, in order to show you other quests around that place.  (it can also be undone as you've been doing if you change your mind and want to solve it after all). 
  The main advantage is that you don't have to undo at all, as such hiding is local to your device, so you won't be uploading any wrong answers even if you don't undo. (Also, there is `Settings` / `Restore hidden quests` if you prefer to show all of them again with one action - for example at the end of mapping session or similar)
- in `Settings` / `Quest selection and display order` you can set so you don't see the quests that you don't want to be solving at all; and it also allows you to drag&drop quest to change their priorities (so for example you can set that `Address quest` is always shown to you before for example `number of floors` quests for the building. (do note that some quests depend on the other, for example until [#2464](https://github.com/streetcomplete/StreetComplete/issues/2464) situation changes, you will have to solve `building type` before you can solve `house address` quest). 
  However, if you're already **only** displaying/solving `housenumber` quest, and one address is hidden by the other, but in the wrong order, this one won't help you much. The first point however (hiding instead of giving wrong answer) is still much preferable (it does not mess up the counter, and it removes the chance wrong data is going to be uploaded).

- Also, since `SC v34` you can use **quests presets** (in `Settings` / `Quest selection and display order` / `... menu` / `manage presets`). This could help you to make a separate preset for **night mapping** that you mention, which would only have quests that you can see / want to solve during the night. And _less quests_ means _less crowding on the screen_  which means _less of the quests get hidden behind a dot_.


Other than that (or solving all the quests around!), there is not much help at the moment, I think. If you are maximally zoomed in already, the quests behind a dot (or even hidden altogether if there are many dots!) are not solvable, due to underlying library (TangramES) being used. There have been some proposed changes that might change that (https://github.com/streetcomplete/StreetComplete/issues/3123 or https://github.com/streetcomplete/StreetComplete/issues/124#issuecomment-716066826 come to mind)

**Reply by timothywashere** [reply]
Thanks for the comprehensive reply :)

Hiding is probably a better answer than fake answers.

The problem with hiding the number quest, is that that one I've hidden is usually the one I want to solve next.

**Reply by mnalis** [reply]
Yes, if using for example only `house address` quest, that is a problem by not being able too zoom in enough: https://github.com/tangrams/tangram-es/issues/2264 (see also https://github.com/streetcomplete/StreetComplete/issues/2359 and https://github.com/streetcomplete/StreetComplete/issues/2878). No good workaround for that specific use case currently, AFAIK.
